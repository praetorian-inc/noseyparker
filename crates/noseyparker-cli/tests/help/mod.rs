//! Tests for Nosey Parker `help` functionality

use super::*;

#[test]
fn no_args() {
    assert_cmd_snapshot!(noseyparker!().assert().failure());
}

#[test]
fn help() {
    assert_cmd_snapshot!(noseyparker_success!("help"));
}

#[test]
fn help_short() {
    assert_cmd_snapshot!(noseyparker_success!("-h"));
}

#[test]
fn help_scan() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "scan"));
    });
}

#[test]
fn help_scan_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("scan", "-h"));
    });
}

#[test]
fn help_summarize() {
    assert_cmd_snapshot!(noseyparker_success!("help", "summarize"));
}

#[test]
fn help_summarize_short() {
    assert_cmd_snapshot!(noseyparker_success!("summarize", "-h"));
}

#[test]
fn help_report() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "report"));
    });
}

#[test]
fn help_report_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("report", "-h"));
    });
}

#[test]
fn help_datastore() {
    assert_cmd_snapshot!(noseyparker_success!("help", "datastore"));
}

#[test]
fn help_rules() {
    assert_cmd_snapshot!(noseyparker_success!("help", "rules"));
}

#[test]
fn help_github() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github"));
}

#[test]
fn help_github_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "-h"));
}

#[test]
fn help_github_repos() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github", "repos"));
}

#[test]
fn help_github_repos_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "repos", "-h"));
}

// #[test]
// fn version_short() {
//     assert_cmd_snapshot!(noseyparker_success!("-V"));
// }

#[test]
fn version_long() {
    with_settings!({
        filters => vec![
            (r"(?m)^(    [^:]+:[ \t]+).*$", r"$1<PLACEHOLDER>")
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("--version"));
    });
}

#[test]
fn version_command() {
    // there is no `version` command
    assert_cmd_snapshot!(noseyparker_failure!("version"));
}
