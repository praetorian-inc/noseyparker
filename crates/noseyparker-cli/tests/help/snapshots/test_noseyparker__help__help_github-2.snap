---
source: crates/noseyparker-cli/tests/help/mod.rs
expression: stdout
---
Interact with GitHub

By default, unauthenticated access is used. An optional personal access token can be specified using
the `NP_GITHUB_TOKEN` environment variable. Using a personal access token gives higher rate limits
and may make additional content accessible.

Usage: noseyparker github [OPTIONS] <COMMAND>

Commands:
  repos  Interact with GitHub repositories
  help   Print this message or the help of the given subcommand(s)

Options:
      --github-api-url <URL>
          Use the specified URL for GitHub API access
          
          If accessing a GitHub Enterprise Server instance, this value should be the entire base URL
          include the `api/v3` portion, e.g., `https://github.example.com/api/v3`.
          
          [default: https://api.github.com/]
          [aliases: api-url]

  -h, --help
          Print help (see a summary with '-h')

Global Options:
  -v, --verbose...
          Enable verbose output
          
          This can be repeated up to 3 times to enable successively more output.

      --color <MODE>
          Enable or disable colored output
          
          When this is "auto", colors are enabled when stdout is a tty.
          
          If the `NO_COLOR` environment variable is set, it takes precedence and is equivalent to
          `--color=never`.
          
          [default: auto]
          [possible values: auto, never, always]

      --progress <MODE>
          Enable or disable progress bars
          
          When this is "auto", progress bars are enabled when stderr is a tty.
          
          [default: auto]
          [possible values: auto, never, always]

Advanced Global Options:
      --rlimit-nofile <LIMIT>
          Set the rlimit for number of open files to LIMIT
          
          This should not need to be changed from the default unless you run into crashes from
          running out of file descriptors.
          
          [default: 16384]

      --enable-backtraces <BOOL>
          Enable or disable backtraces on panic
          
          This has the effect of setting the `RUST_BACKTRACE` environment variable to 1.
          
          [default: true]
          [possible values: true, false]

